src/axom/primal/operators/closest_point.hpp

```diff
@@ -23,6 +24,50 @@ namespace axom
 {
 namespace primal
 {
+
+/*!
+ * \brief Computes the closest point from a point, P, to a given segment.
+ *
+ * \param [in] P the query point
+ * \param [in] seg user-supplied segment
+ * \param [in] EPS fuzz factor for equality comparisons
+ * \return cp the closest point from a point P and a segment
+ */
+template <typename T, int NDIMS>
+AXOM_HOST_DEVICE inline Point<T, NDIMS> closest_point(const Point<T, NDIMS>& P,
+                                                      const Segment<T, NDIMS>& seg,
+                                                      double EPS = 1E-8) {
```

Comment:
It might also be useful to return the `t` value as an out parameter, which users could ignore if they don't need it. 

It shouldn't have any overhead since we're already computing it.

Reply:
Done.